File: kyubyungchae/0x05/10828.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {

    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    stack<int> st;
    while(n--) {
        string c;
        cin >> c;
        if(c == "push") {
            int t;
            cin >> t;
            st.push(t);
        } else if(c == "pop") {
            if(st.empty()) 
                cout << -1 << '\n';
            else {
                cout << st.top() << '\n';
                st.pop();
            }
        } else if(c == "size") {
            cout << st.size() << '\n';
        } else if(c == "empty") {
            cout << (int)st.empty() << '\n';
        } else {
            if(st.empty()) 
                cout << -1 << '\n';
            else 
                cout << st.top() << '\n';
        }
    }


    return 0;
}

File: sangsoo/0x05 stack/10828.py
# BOJ 10828 스택
# input
import sys

sys.stdin = open('../bojinput/10828.txt', 'r')

n = int(input())
order = []
for i in range(n):
    order.append(input())

stack = []
for i in range(n):
    orderstring = order[i].split(' ')[0]
    # print(f"orderstring:{orderstring}")
    if orderstring == 'push':
        stack.append(order[i].split(' ')[1])
    elif orderstring == 'pop':
        if not stack:
            print(-1)
        else:
            print(stack.pop())
    elif orderstring == 'size':
        print(len(stack))
    elif orderstring == 'empty':
        if not stack:
            print(1)
        else:
            print(0)
    elif orderstring == 'top':
        if not stack:
            print(-1)
        else:
            print(stack[-1])

